lib/python/platform_code_vocab.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Find the platform_code platform_name equivalence from poolparty platform vocab
+xml file stored in content.aodn.org.au
+
+How to use:
+    from platform_code_vocab import *
+
+    platform_altlabels_per_preflabel()
+    platform_type_uris_by_category()
+    platform_altlabels_per_preflabel('Fixed station')
+    platform_altlabels_per_preflabel('Mooring and buoy')
+    platform_altlabels_per_preflabel('Vessel')
+
+author : Besnard, Laurent
+"""
+
+import urllib2
+import warnings
+import xml.etree.ElementTree as ET
+
+
+def platform_type_uris_by_category():
+    """
+    retrieves a list of platform category and their narrowMatch url type which
+    defines their category
+    """
+    platform_cat_vocab_url = 'http://content.aodn.org.au/Vocabularies/platform-category/aodn_aodn-platform-category-vocabulary.rdf'
+    response               = urllib2.urlopen(platform_cat_vocab_url)
+    html                   = response.read()
+    root                   = ET.fromstring(html)
+    platform_cat_list      = {}
+
+    for item in root:
+        if 'Description' in item.tag:
+            platform_cat_url_list = []
+            platform_cat          = None
+
+            for val in item:
+                platform_element_sublabels = val.tag
+
+                # handle more than 1 url match per category of platform
+                if platform_element_sublabels is not None:
+                    if 'narrowMatch' in platform_element_sublabels:
+                        val_cat_url = val.attrib[val.attrib.keys()[0]]
```

Comment:
A simpler way would be `val_cat_url = val.attrib.values()[0]`. Also you have to be careful addressing `dict` objects by index because they are not ordered! If `val.attrib` has more than one item, this will give you a random one. Is that what you want?